src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -73,201 +76,138 @@ public RoslynAnalyzerProvider(IAnalyzerInstaller analyzerInstaller, ILogger logg
             {
                 throw new ArgumentNullException(nameof(inactiveRules));
             }
+
             if (activeRules == null)
             {
                 throw new ArgumentNullException(nameof(activeRules));
             }
 
-            var analyzer = ConfigureAnalyzer(language, activeRules, inactiveRules);
-            if (analyzer == null)
-            {
-                this.logger.LogInfo(Resources.RAP_NoPluginInstalled, language);
-            }
+            var rulesetFilePath = CreateRuleSet(language, activeRules, inactiveRules, ProjectType.Product);
+            var testProjectRuleSetFilePath = CreateRuleSet(language, activeRules, inactiveRules, ProjectType.Test);
+            var analyzersAssemblies = FetchAnalyzerAssemblies(language, activeRules);
+            var additionalFiles = WriteAdditionalFiles(language, activeRules);
 
-            return analyzer;
+            return new AnalyzerSettings(language, rulesetFilePath, testProjectRuleSetFilePath, analyzersAssemblies, additionalFiles);
```

Comment:
Previously the method would return `null` if there weren't any rules for a language (it looks like one of the cases this was intended to handle was when a plugin wasn't installed. This would normally have been the VB plugin, I guess).

The method now never returns null. What impact does this have on the callers?

Reply:
ConfigureAnalyzer returns null when WriteRuleSet returns null
WriteRuleSet returns null when Generate returned null or Rules was null.
Generate never returned null and the Rules collection was never null because it is initialized with new List<> and the property is never set with null.

See https://github.com/SonarSource/sonar-scanner-msbuild/pull/690/files/1c2fd8c3397299fd7d02d5cda615a4cd2672d9e4#diff-d878b31e98e7b201477d4362bfe06cd1L44

Reply:
As far as I could tell it is creating different AnalyzerSettings for each plugin, hence if a plugin is not installed we will not create AnalyzerSettings...